File: 프로그래머스/lv1/12925. 문자열을 정수로 바꾸기/문자열을 정수로 바꾸기.rb
  def solution(str)
    str_num = ""
    oper = true
    str.each_char do |c|
      if c == '-'
        oper = false
      elsif c == '+'
        next
      else
        str_num += c
      end
    end

    num = str_num.to_i
    oper ? num : num * -1
  end

File: 프로그래머스/lv1/12925. 문자열을 정수로 바꾸기/문자열을 정수로 바꾸기.java
class Solution {
    public int solution(String str) {
        String strNum = "";
        boolean oper = true;
        for(char c:str.toCharArray()){
            if(c == '-'){
                oper = false;
            }else if(c == '+'){
                continue;
            }else{
                strNum += c;
            }
        }

        int num = Integer.parseInt(strNum);
        return oper?num:num*-1;
    }
}